torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1728,12 +1736,29 @@ def _local_state_dict(self, *args: Any, **kwargs: Any) -> Any:
         with self.state_dict_type(self, StateDictType.LOCAL_STATE_DICT):
             return self.state_dict(*args, **kwargs)
 
+    def _full_post_load_state_dict_hook(self, *args, **kwargs) -> None:
+        # We should exit summon_full_params context.
+        assert getattr(self, '_entered_full_param_ctx', None) is not None
+        self._entered_full_param_ctx.__exit__(None, None, None)
+        self._entered_full_param_ctx = None
+
     def _full_pre_load_state_dict_hook(
         self,
         state_dict: Dict[str, Any],
         prefix: str,
     ) -> None:
         _replace_by_prefix(state_dict, prefix, prefix + f"{FSDP_WRAPPED_MODULE}.")
+        # We do not expect to be calling pre-hooks twice without post-hook
+        # call in between.
+        assert getattr(self, '_entered_full_param_ctx', None) is None
+        # Note that it needs writeback=True to persist.
+        self._entered_full_param_ctx = self._summon_full_params(
```

Comment:
It looks like `_entered_full_param_ctx` is the context manager itself?
In that case, do you think it would be clearer to call it `_full_param_ctx`?

`_entered_full_param_ctx` seems more like `_full_param_ctx is not None`.